microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/PropertiesConfigSource.java

```diff
@@ -46,7 +47,10 @@ public PropertiesConfigSource(String configFileName, int ordinal, IJavaProject j
 	@Override
 	public String getProperty(String key) {
 		Properties properties = getConfig();
-		return properties != null ? properties.getProperty(key) : null;
+		if (properties == null) {
+			return null;
+		}
+		return new PropertyValueExpander(properties).getValue(key);

```

Comment:
I think expansion should be done: 

 * only if value is not an expression (the value doesn't contains `${`).
 * done just one time. Perhaps you could store the expanded value in the properties instance once it is expaned (but what about when expansion throw an error?)
 * it requires some test, but expand should perhaps use the scope of properties file (application.properties and meta-config.properties file)?

Reply:
Okay. It should now work with multiple properties files. (i.e. it can resolve properties that are only given a value in a different property file). I had to modify the API for IConfigSource (there wasn't a way around doing it that was obvious to me).

Reply:
It seems like this API change doesn't affect quarkus-ls